terraform/groups/provisioning-role/module-iam-policies/terraform-destroy.tf

```diff
@@ -0,0 +1,47 @@
+data "aws_iam_policy_document" "destroy" {
+  statement {
+    sid       = "DestroyPolicyListedResourcesWithConditions"
+    effect    = "Allow"
+    resources = [
+      "arn:aws:ec2:eu-west-2:${var.aws_account_id}:security-group/*"
+    ]
+    actions   = [
+      "ec2:DeleteSecurityGroup"
+    ]
+    condition {
+      test     = "ForAllValues:StringLike"
+      variable = "ec2:ResourceTag/Name"
+      values   = [
+        "*-payment-reconciler-sg"
+      ]
+    }
+  }
+
+  statement {
+    sid       = "DestroyPolicyAllResources"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = [
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DeleteNetworkInterface",
+    ]
+  }
```

Comment:
Would this allow deletion of ANY network interfaces?

Reply:
I think so. I spent some time yesterday trying to find a way to restrict it - I couldn't find a way.
The ENI is created automatically I think from the Lambda VPC configuration and has no name and no tag. I did look to see if we could create the ENI separately in Terraform (which we could) but the aws_lambda_function resource doesn't allow you to then specify this - so stuck with letting it create automatically. If it's not tagged and not named, I couldn't think of a way to add any condition on it here.
Any ideas?